exporter/lokiexporter/config.go

```diff
@@ -32,19 +32,41 @@ type Config struct {
 	exporterhelper.RetrySettings  `mapstructure:"retry_on_failure"`
 
 	// TenantID defines the tenant ID to associate log streams with.
+	// Deprecated: use Tenant instead, with Source set to "static" and
+	// Value set to the value you'd set in this field.
 	TenantID string `mapstructure:"tenant_id"`
 
 	// Labels defines how labels should be applied to log streams sent to Loki.
 	Labels LabelsConfig `mapstructure:"labels"`
+
 	// Allows you to choose the entry format in the exporter
 	Format string `mapstructure:"format"`
+
+	// Tenant defines how to obtain the tenant ID
+	Tenant *Tenant `mapstructure:"tenant"`
+}
+
+type Tenant struct {
+	// Source defines where to obtain the tenant ID. Possible values: static, context, attribute.
+	Source string `mapstruct:"source"`
+
+	// Value will be used by the tenant source provider to lookup the value. For instance,
+	// when the source=static, the value is a static value. When the source=context, value
+	// should be the context key that holds the tenant information.
+	Value string `mapstruct:"value"`
 }
 
 func (c *Config) validate() error {
 	if _, err := url.Parse(c.Endpoint); c.Endpoint == "" || err != nil {
 		return fmt.Errorf("\"endpoint\" must be a valid URL")
 	}
 
+	if c.Tenant != nil {
```

Comment:
What happens if both `Tenant` and `TenantID` are used at the same time?

Reply:
I like failing.  Maybe a little harsh, but mixing deprecated fields and their replacement fields feels like something we should help our users avoid.

Reply:
PR updated